File: Python/Classic Puzzles/Easy/the-descent.py
# Initialize an empty list to store the heights of the mountains
mountain_h = []

# Start an infinite loop to continuously process mountain heights
while True:
    # Clear the list for each new iteration (to avoid storing old data)
    mountain_h = []

    # Read 8 mountain heights from input and store them in the list
    for i in range(8):
        mountain_h.append(int(input()))

    # Find the index of the tallest mountain in the list
    index_max = mountain_h.index(max(mountain_h))

    # Print the index of the tallest mountain
    print(index_max)

File: JavaScript/Classic Puzzles/Easy/the-descent.js
let mountainH = []

// game loop
while (true) {

    mountainH = []
    
    for (let i = 0; i < 8; i++) {
        mountainH.push(parseInt(readline())); // represents the height of one mountain.
    }

    const maxHeight = Math.max(...mountainH); // Spread operator to pass array elements
    const indexMax = mountainH.indexOf(maxHeight); // Find the index of the max height

    console.log(indexMax);     // The index of the mountain to fire on.

}
